### PR TITLE
fix: allow using --select-nth with --continue

### PR DESCRIPTION
**Pull request by FireEgl**
Allows using --select-nth with --continue to select the anime when there's multiple to choose from.

# Pull Request Template

## Type of change

- [x] Bug fix
- [x] Feature
- [ ] Documentation update

## Description

This adds a check for --select-nth when using --continue to select the anime, so it can be done non-interactively.
Updated to work with the current code base. =)

## Checklist

- [ ] any anime playing
- [ ] bumped version
---
- [ ] next, prev and replay work
- [x] `-c` history and continue work
- [x] `-d` downloads work
- [ ] `-s` syncplay works
- [ ] `-q` quality works
- [ ] `-v` vlc works
- [ ] `-e` select episode works
- [x] `-S` select index works
- [ ] `-r` range selection works
- [ ] `--skip` ani-skip works
- [ ] `--skip-title` ani-skip title argument works
- [ ] `--no-detach` no detach works
- [ ] `--dub` and regular (sub) mode both work
- [ ] all providers return links (not necessarily on a single anime, use debug mode to confirm)
---
- [ ] `-h` help info is up to date
- [ ] Readme is up to date
- [ ] Man page is up to date

## Additional Testcases

- The safe bet: One Piece
- Episode 0: Saenai Heroine no Sodatekata ♭
- Unicode: Saenai Heroine no Sodatekata ♭
- Non-whole episodes: Tensei shitara slime datta ken (ep. 24.5, ep. 24.9)

### CODE REVIEW

**Comment by 71zenith** [comment]
as i said before, u cant even predict which index will have which anime, making this useless

**Comment by FireEgl** [comment]
> as i said before, u cant even predict which index will have which anime, making this useless

Yes, it's unpredictable which anime it downloads/plays. But it also doesn't matter.  I just want to periodically run the script to have all my animes downloaded (ani-cli -d -c -S 1).  I get that it kind of mis-uses what --select-nth was meant for, but it works.
The alternative is to rework a lot of the script to have a properly working non-interactive mode.. But since this might be the only use-case, I don't see the problem with a little bit of a hack.

**Comment by port19x** [comment]
Valid use case for a tiny change

**Review by port19x — COMMENTED** [review]
I wonder how this code doesn't trigger [SC2015](https://www.shellcheck.net/wiki/SC2015)

**Comment by FireEgl** [comment]
> I wonder how this code doesn't trigger [SC2015](https://www.shellcheck.net/wiki/SC2015)

I tried checking the same code without the id= part and shellcheck gives a SC2015 (info) on it..
So, if somehow the interactive prompt fails, it's going to try to set it using --select-nth which will also fail, and the next line is a `[ -z "$id" ] && exit 1` anyway, so it's fine. =)

File: ani-cli [hunk 1]
```diff
@@ -427,7 +427,8 @@ case "$search" in
         anime_list=$(while read -r ep_no id title; do process_hist_entry & done <"$histfile")
         wait
         [ -z "$anime_list" ] && die "No unwatched series in history!"
-        id=$(printf "%s" "$anime_list" | nl -w 2 | sed 's/^[[:space:]]//' | nth "Select anime: " | cut -f1)
+        # Prompts for the anime, or selects based on the index (--select-nth) value if provided:
+        [ -z "${index##*[!0-9]*}" ] && id=$(printf "%s" "$anime_list" | nl -w 2 | sed 's/^[[:space:]]//' | nth "Select anime: " | cut -f1) || id=$(printf "%s" "$anime_list" | sed -n "${index}p" | cut -f1)
```
Comment (port19x) [inline ani-cli:431]: I would have expected the logic to go like this, but maybe I'm missing something

```suggestion
        [ -z "${index##*[!0-9]*}" ] && id=$(printf "%s" "$anime_list" | nl -w 2 | sed 's/^[[:space:]]//' | nth "Select anime: " | cut -f1) 
        [ -z "${index##*[!0-9]*}" ] || id=$(printf "%s" "$anime_list" | sed -n "${index}p" | cut -f1)
```

**Comment by 71zenith** [comment]
`${index##*[!0-9]*}` very cool way to type check and it seems this is posix as well.
might want to incorporate more into the codebase rather than `-eq`

File: ani-cli [hunk 2]
```diff
@@ -427,7 +427,8 @@ case "$search" in
         anime_list=$(while read -r ep_no id title; do process_hist_entry & done <"$histfile")
         wait
         [ -z "$anime_list" ] && die "No unwatched series in history!"
-        id=$(printf "%s" "$anime_list" | nl -w 2 | sed 's/^[[:space:]]//' | nth "Select anime: " | cut -f1)
+        # Prompts for the anime, or selects based on the index (--select-nth) value if provided:
```
[the remark this answers is not in the document]
Comment (FireEgl) [reply ani-cli:430]: I dunno how to do this without making another branch and pull-request like you guys, but what about this:
```
--- ani-cli	2024-06-10 07:55:18.846673971 -0500
+++ ani-cli.fireegl-2	2024-06-11 08:21:47.636001449 -0500
@@ -427,7 +427,10 @@
         anime_list=$(while read -r ep_no id title; do process_hist_entry & done <"$histfile")
         wait
         [ -z "$anime_list" ] && die "No unwatched series in history!"
-        id=$(printf "%s" "$anime_list" | nl -w 2 | sed 's/^[[:space:]]//' | nth "Select anime: " | cut -f1)
+        case "$index" in
+           ''|*[!0-9]*) id=$(printf "%s" "$anime_list" | nl -w 2 | sed 's/^[[:space:]]//' | nth "Select anime: " | cut -f1) ;;
+           *) id=$(printf "%s" "$anime_list" | sed -n "${index}p" | cut -f1) ;;
+        esac
         [ -z "$id" ] && exit 1
         title=$(printf "%s" "$anime_list" | grep "$id" | cut -f2 | sed 's/ - episode.*//')
         ep_list=$(episodes_list "$id")
```

[the remark this answers is not in the document]
Comment (71zenith) [reply ani-cli:430]: dont use `case`. to maintain parity with rest of the codebase, just check it twice like port did. not very efficient but meh who cares.

[the remark this answers is not in the document]
Comment (71zenith) [reply ani-cli:430]: just pull the changes ? i assume u havent pulled the version bump port did.
and make a new commit

[the remark this answers is not in the document]
Comment (71zenith) [reply ani-cli:430]: dont open a new pr please

[the remark this answers is not in the document]
Comment (71zenith) [reply ani-cli:430]: if u cannot do this, i can make the changes

[the remark this answers is not in the document]
Comment (FireEgl) [reply ani-cli:430]: Please do. =)

[the remark this answers is not in the document]
Comment (71zenith) [reply ani-cli]: done, if ure having issues u can just use the github editor line . just open the file on gh and use the pen button to edit there

[the remark this answers is not in the document]
Comment (71zenith) [reply ani-cli]: port can handle the rest

**Comment by port19x** [comment]
@71zenith you sure we wanna bump to 4.9?

**Comment by port19x** [comment]
We do have countdown to next episode as a feature I guess

**Comment by 71zenith** [comment]
i believe we have enough fixes accumulated? did we aim at having some milestones for each release?
just thought that .10 felt wrong. if u want we can wait a bit for the logger feature

**Comment by port19x** [comment]
Let's wait for the logger feature then. 
Almost all our downstreams being up-to-date is just very satisfying to look at on repology, so I'm hesitant.
But you're right, we have a lot of small bumps and features

**Review by port19x — APPROVED** [review]

**Comment by port19x** [comment]
@71zenith you're gonna approve or no?

**Review by 71zenith — APPROVED** [review]

**Comment by 71zenith** [comment]
approved, sorry i was at school

**Comment by KuraiZ0** [comment]
I like your work !
I havé a suggestions if possible, add others languages like French, Spain Will give you More visibility !

**Pull request merged**